titan/features/haart.py

```diff
@@ -99,18 +101,36 @@ def update_agent(self, model: "model.TITAN"):
                     ]
                     num_haart_agents = self.counts[self.agent.race][self.agent.sex_type]
 
-                    if num_haart_agents < (haart_params.prob * num_dx_agents):
+                    # take value from dictionary for cap
+                    if num_haart_agents < (haart_params.cap * num_dx_agents):
                         self.initiate(model)
                 else:
-                    if model.run_random.random() < (
-                        haart_params.prob * model.calibration.haart.coverage
-                    ):
-                        self.initiate(model)
+                    if self.ever and self.agent.location.params.hiv.reinit_haart:
+                        if model.run_random.random() < haart_params.reinit.prob:
+                            self.initiate(model)
+                    else:
+                        # Find enroll probability based on time since diagnosis
+                        for i in haart_params.prob.values():
+                            if i.start <= (model.time - self.agent.hiv.dx_time) < i.stop:  # type: ignore[attr-defined]
+                                enroll_prob = i.enroll_prob
+                                break
+
+                        if model.run_random.random() < (
+                            enroll_prob * model.calibration.haart.coverage
+                        ):
+                            self.initiate(model)
+
             # Go off HAART
-            elif self.active and model.run_random.random() < haart_params.discontinue:
-                self.active = False
-                self.adherent = False
-                self.remove_agent(self.agent)
+            elif self.active:
+                if model.run_random.random() < haart_params.discontinue:
+                    self.active = False
+                    self.adherent = False
+                    self.remove_agent(self.agent)
+                elif (
+                    self.adherent
+                    and model.run_random.random() < haart_params.adherence.discontinue
+                ):
```

Comment:
is there a way for an agent to go from non-adherent to adherent?

Reply:
Uhhhhh, there will be!